dietpi/dietpi-software

```diff
@@ -12091,6 +12098,73 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
 
 			unset -v choice_required domain port invalid_text server_ip
 		fi
+
+		if To_Install 213 # soju
+		then
+			local ver=0.8.0
+			aDEPS=('gcc' 'make' 'pkg-config' 'scdoc' 'libpam-dev' 'libsqlite3-dev')
+			Download_Install "https://git.sr.ht/~emersion/soju/archive/v${ver}.tar.gz"
+			G_EXEC cd "soju-v${ver}"
+
+			# Build
+			G_EXEC_OUTPUT=1 G_EXEC make all "-j$(nproc)" \
+				config_path=/mnt/dietpi_userdata/soju/config
+				GOFLAGS=-tags=pam,libsqlite3 \
+
+			# Install
+			local commands=('soju' 'sojuctl' 'sojudb')
+			local manpages=('doc/soju.1' 'doc/sojuctl.1')
+			G_EXEC mkdir -p /usr/local/bin
+			G_EXEC mkdir -p /usr/local/share/man/man1
+			G_EXEC mkdir -p /mnt/dietpi_userdata/soju
+			G_EXEC cp "${commands[@]}" /usr/local/bin
+			G_EXEC cp "${manpages[@]}" /usr/local/share/man/man1
+			if [[ ! -f '/mnt/dietpi_userdata/soju/config' ]]; then
+				# We shall default to on-disk chat logs
+				# It's a lossy format, as stated under `message-store` subsection in https://soju.im/doc/soju.1.html#CONFIG_FILE
+				# But it's the recommended setting as per https://git.sr.ht/~emersion/soju/tree/master/doc/getting-started.md
+				G_EXEC touch /mnt/dietpi_userdata/soju/config
+				cat <<- _EOF_ > /mnt/dietpi_userdata/soju/config
+				db sqlite3 /mnt/dietpi_userdata/soju/data.db
+				message-store fs /mnt/dietpi_userdata/soju/logs
+				listen irc+insecure://
+				listen unix+admin://
+				_EOF_
+			fi
+
+			# User
+			Create_User soju
+			G_EXEC chown soju:soju -R /mnt/dietpi_userdata/soju
+			G_EXEC chmod 0660 -R /mnt/dietpi_userdata/soju
+			G_EXEC chmod 0770 /mnt/dietpi_userdata/soju
+
+			# Service
+			# Modified version of ./contrib/soju.service
+			cat << _EOF_ > /etc/systemd/system/soju.service
+[Unit]
+Description=soju IRC bouncer service
+Documentation=https://soju.im/
+Documentation=man:soju(1) man:sojuctl(1)
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+Type=simple
+User=soju
+Group=soju
+RuntimeDirectory=soju
+RuntimeDirectoryMode=0770
+RuntimeDirectoryPreserve=restart
+WorkingDirectory=/mnt/dietpi_userdata/soju
+AmbientCapabilities=CAP_NET_BIND_SERVICE
```

Comment:
Does it listen on other ports than 6667? Because `CAP_NET_BIND_SERVICE` is only required for privileged ports <1024.

Reply:
Here is the explanation: https://github.com/emersion/soju/commit/662e3f3
It has a builtin `identd` server: https://github.com/emersion/soju/blob/master/doc/packaging.md#binding-to-privileged-ports